src/QirRuntime/public/ResourceStatistics.hpp

```diff
@@ -0,0 +1,55 @@
+#pragma once
+#include <sstream>
+
+namespace Microsoft
+{
+namespace Quantum
+{
+    static const char* names[] = {"cX", "cCX", "cY",  "cCY",  "cZ", "cCZ", "cH",  "cCH",  "cR", "cCR",
+                                  "cS", "cCS", "cAS", "cCAS", "cT", "cCT", "cAT", "cCAT", "cM"};
+    static constexpr int counters = sizeof(names) / sizeof(char*);
+
+    struct ResourceStatistics
+    {
+        // these members must match names[] above

```

Comment:
Why not just use a `map<string, long>` so the code isn't so fragile and limited?